python/runtime/db.py

```diff
@@ -192,29 +192,97 @@ def read_feature(raw_val, feature_spec, feature_name):
         return raw_val,
 
 
-def selected_cols(conn, select):
+LIMIT_PATTERN = re.compile("LIMIT\\s+([0-9]+)", flags=re.I)
+
+
+def limit_select(select, n):
+    """Make the select SQL statement with limited row number to query.
+
+    Args:
+        select (str): the select SQL statement.
+        n (int): the limited row number to query.
+
+    Returns:
+        If n >= 0, return a new SQL statement which would query n row(s) at most.
+        If n < 0, return the original SQL statement.
+    """
+    if n < 0:
+        return select
+
+    def replace_limit_num(matched_limit):
+        num = int(matched_limit.group(1))
+        return "LIMIT {}".format(min(num, n))
+
+    if LIMIT_PATTERN.search(select) is None:
+        return select + " LIMIT {}".format(n)
+    else:
+        return LIMIT_PATTERN.sub(repl=replace_limit_num, string=select)
+
+
+MYSQL_DATA_TYPE_DICT = {
+    1: "TINYINT",
+    3: "INT",
+    4: "FLOAT",
+    5: "DOUBLE",
+    8: "BIGINT",
+    246: "DECIMAL",
+    252: "TEXT",
+    253: "VARCHAR",
+    254: "CHAR",
```

Comment:
How about use MySQLdb.constants.FIELD_TYPE constants instead of the numbers?

Reply:
Thanks for your suggestions! Would fix it in the next PR.